examples/rust/tcp-push-pop.rs

```diff
@@ -154,7 +154,8 @@ impl TcpServer {
         self.accepted_qd = match self.libos.wait(qt, None) {
             Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_ACCEPT => unsafe { Some(qr.qr_value.ares.qd.into()) },
             Err(e) => anyhow::bail!("operation failed: {:?}", e.cause),
-            Ok(_) => anyhow::bail!("unexpected result"),
+            Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_FAILED => anyhow::bail!("accept failed: {}", qr.qr_ret),
```

Comment:
It would be good to keep the `Err` arm last, just for ease of reading the code. (applies for other such places as well)